### ISSUE TITLE
Panics with a wgpu validation error

### ISSUE THREAD

**Issue by yusdacra** [hecrj/iced#738]
I recently updated a project of mine to use the latest `master` git commit. This updated `wgpu` from `0.6.2` to `0.7`. After I built and ran my application I started to get this panic:
```
wgpu error: Validation Error

Caused by:
    In a RenderPass
      note: encoder = `iced_wgpu encoder`
    In a set_scissor_rect command
    Invalid ScissorRect parameters


thread 'main' panicked at 'Handling wgpu errors as fatal by default'
```

It was working fine before. `vkcube` and `vulkaninfo` are both working and other applications which use Vulkan also work.

<details>
<summary>Log</summary>

```
18:35:47 [WARN] gfx_backend_vulkan: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/gfx-backend-vulkan-0.7.0/src/lib.rs:412] Unable to find layer: VK_LAYER_KHRONOS_validation
18:35:47 [WARN] gfx_backend_vulkan: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/gfx-backend-vulkan-0.7.0/src/lib.rs:971] Skipping memory type with unknown flags DEVICE_LOCAL | DEVICE_COHERENT_AMD | DEVICE_UNCACHED_AMD
18:35:47 [WARN] gfx_backend_vulkan: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/gfx-backend-vulkan-0.7.0/src/lib.rs:971] Skipping memory type with unknown flags HOST_VISIBLE | HOST_COHERENT | DEVICE_COHERENT_AMD | DEVICE_UNCACHED_AMD
18:35:47 [WARN] gfx_backend_vulkan: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/gfx-backend-vulkan-0.7.0/src/lib.rs:971] Skipping memory type with unknown flags DEVICE_LOCAL | HOST_VISIBLE | HOST_COHERENT | DEVICE_COHERENT_AMD | DEVICE_UNCACHED_AMD
18:35:47 [WARN] gfx_backend_vulkan: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/gfx-backend-vulkan-0.7.0/src/lib.rs:971] Skipping memory type with unknown flags HOST_VISIBLE | HOST_COHERENT | HOST_CACHED | DEVICE_COHERENT_AMD | DEVICE_UNCACHED_AMD
18:35:47 [WARN] naga::front::spv: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/naga-0.3.1/src/front/spv/mod.rs:440] Unknown decoration ColMajor
18:35:47 [WARN] naga::front::spv: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/naga-0.3.1/src/front/spv/mod.rs:440] Unknown decoration MatrixStride
18:35:47 [WARN] naga::front::spv: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/naga-0.3.1/src/front/spv/mod.rs:2365] Treating VertexIndex as unsigned
18:35:47 [WARN] naga::front::spv: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/naga-0.3.1/src/front/spv/mod.rs:440] Unknown decoration ColMajor
18:35:47 [WARN] naga::front::spv: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/naga-0.3.1/src/front/spv/mod.rs:440] Unknown decoration MatrixStride
18:35:47 [WARN] naga::front::spv: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/naga-0.3.1/src/front/spv/mod.rs:1552] Unused item decorations: {13: Decoration { name: Some("size"), built_in: None, location: None, desc_set: None, desc_index: None, block: None, offset: None, array_stride: None, interpolation: None, flags: (empty) }, 14: Decoration { name: Some("radius"), built_in: None, location: None, desc_set: None, desc_index: None, block: None, offset: None, array_stride: None, interpolation: None, flags: (empty) }, 11: Decoration { name: Some("frag_coord"), built_in: None, location: None, desc_set: None, desc_index: None, block: None, offset: None, array_stride: None, interpolation: None, flags: (empty) }, 12: Decoration { name: Some("position"), built_in: None, location: None, desc_set: None, desc_index: None, block: None, offset: None, array_stride: None, interpolation: None, flags: (empty) }}
18:35:47 [WARN] naga::front::spv: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/naga-0.3.1/src/front/spv/mod.rs:440] Unknown decoration ColMajor
18:35:47 [WARN] naga::front::spv: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/naga-0.3.1/src/front/spv/mod.rs:440] Unknown decoration MatrixStride
18:35:47 [WARN] naga::front::spv: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/naga-0.3.1/src/front/spv/mod.rs:2365] Treating VertexIndex as unsigned
18:35:47 [WARN] naga::front::spv: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/naga-0.3.1/src/front/spv/mod.rs:440] Unknown decoration ColMajor
18:35:47 [WARN] naga::front::spv: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/naga-0.3.1/src/front/spv/mod.rs:440] Unknown decoration MatrixStride
18:35:47 [INFO] winit::platform_impl::platform::x11::window: [/home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/x11/window.rs:140] Guessed window scale factor: 1.0833333333333333
wgpu error: Validation Error

Caused by:
    In a RenderPass
      note: encoder = `iced_wgpu encoder`
    In a set_scissor_rect command
    Invalid ScissorRect parameters


thread 'main' panicked at 'Handling wgpu errors as fatal by default', /home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.7.0/src/backend/direct.rs:1896:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
zsh: segmentation fault (core dumped)  cargo r
```

</details>

**Comment by MoAlyousef** [comment]
Seems related to
gfx-rs/wgpu-rs#762

**Comment by hecrj** [comment]
Could you provide a backtrace?

**Comment by yusdacra** [comment]
I removed my wgpu 0.7 revert commit to get a backtrace... and it seems to work fine now. Not sure why, maybe it was a driver update? Everything else is the same.

**Comment by Zizico2** [comment]
I have the exact same problem. Here's my full backtrace: 

```
Finished dev [unoptimized + debuginfo] target(s) in 0.16s
     Running `target/debug/gitui-rs`
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0

wgpu error: Validation Error

Caused by:
    In a RenderPass
      note: encoder = `iced_wgpu encoder`
    In a set_scissor_rect command
    Invalid ScissorRect parameters


thread 'main' panicked at 'Handling wgpu errors as fatal by default', /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.7.1/src/backend/direct.rs:1896:5
stack backtrace:
   0:     0x5619a254f270 - std::backtrace_rs::backtrace::libunwind::trace::h72c2fb8038f1bbee
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/../../backtrace/src/backtrace/libunwind.rs:96
   1:     0x5619a254f270 - std::backtrace_rs::backtrace::trace_unsynchronized::h1e3b084883f1e78c
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/../../backtrace/src/backtrace/mod.rs:66
   2:     0x5619a254f270 - std::sys_common::backtrace::_print_fmt::h3bf6a7ebf7f0394a
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:79
   3:     0x5619a254f270 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2e8cb764b7fe02e7
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:58
   4:     0x5619a2571f4c - core::fmt::write::h7a1184eaee6a8644
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/fmt/mod.rs:1080
   5:     0x5619a254a302 - std::io::Write::write_fmt::haeeb374d93a67eac
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/io/mod.rs:1516
   6:     0x5619a255167d - std::sys_common::backtrace::_print::h1d14a7f6ad632dc8
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:61
   7:     0x5619a255167d - std::sys_common::backtrace::print::h301abac8bb2e3e81
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/sys_common/backtrace.rs:48
   8:     0x5619a255167d - std::panicking::default_hook::{{closure}}::hde0cb80358a6920a
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:208
   9:     0x5619a2551328 - std::panicking::default_hook::h9b1a691049a0ec8f
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:227
  10:     0x5619a2551d61 - std::panicking::rust_panic_with_hook::h2bdec87b60580584
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:577
  11:     0x5619a24afa8c - std::panicking::begin_panic::{{closure}}::hffb3eb13089d0a4e
                               at /home/bernardo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:506
  12:     0x5619a24b0c39 - std::sys_common::backtrace::__rust_end_short_backtrace::h67e403e9b891fc4f
                               at /home/bernardo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys_common/backtrace.rs:153
  13:     0x5619a24af9b7 - std::panicking::begin_panic::h35bb6859a53b6be9
                               at /home/bernardo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:505
  14:     0x5619a145c2f1 - wgpu::backend::direct::default_error_handler::h6473aa3c11fa96e8
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.7.1/src/backend/direct.rs:1896
  15:     0x5619a1575ff8 - core::ops::function::Fn::call::hcd048b6c89eae07f
                               at /home/bernardo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:70
  16:     0x5619a1630617 - <alloc::boxed::Box<F> as core::ops::function::Fn<A>>::call::hc4299dbf03a9545a
                               at /home/bernardo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:1056
  17:     0x5619a145c1c9 - wgpu::backend::direct::ErrorSinkRaw::handle_error::h6b084b9b18d3eeda
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.7.1/src/backend/direct.rs:1883
  18:     0x5619a144e636 - wgpu::backend::direct::Context::handle_error::h286255b8aef6a97c
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.7.1/src/backend/direct.rs:109
  19:     0x5619a145bd42 - <wgpu::backend::direct::Context as wgpu::Context>::command_encoder_end_render_pass::h2c03c0ffe4c867a3
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.7.1/src/backend/direct.rs:1739
  20:     0x5619a15f04ef - <wgpu::RenderPass as core::ops::drop::Drop>::drop::hdadea3d66b6cae52
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.7.1/src/lib.rs:2486
  21:     0x5619a117bc16 - core::ptr::drop_in_place::h0d6c3c22df19b3ee
                               at /home/bernardo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:175
  22:     0x5619a10db01b - iced_wgpu::quad::Pipeline::draw::hc107c710a24d0870
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/iced_wgpu-0.4.0/src/quad.rs:275
  23:     0x5619a10fb1e6 - iced_wgpu::backend::Backend::flush::hd157c54e67c891f2
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/iced_wgpu-0.4.0/src/backend.rs:116
  24:     0x5619a1013d19 - iced_wgpu::backend::Backend::draw::h2198d8614a2c6c2a
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/iced_wgpu-0.4.0/src/backend.rs:82
  25:     0x5619a10a5b6d - <iced_wgpu::window::compositor::Compositor as iced_graphics::window::compositor::Compositor>::draw::h61180a173b70b547
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/iced_wgpu-0.4.0/src/window/compositor.rs:156
  26:     0x5619a10889f9 - iced_winit::application::run_instance::{{closure}}::h62eb2223df7f764e
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/iced_winit-0.3.0/src/application.rs:350
  27:     0x5619a10a67e9 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::ha18053d9beae4e7f
                               at /home/bernardo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/future/mod.rs:80
  28:     0x5619a108b22e - iced_winit::application::run::{{closure}}::h344d72aaeab75ac3
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/iced_winit-0.3.0/src/application.rs:193
  29:     0x5619a10ad443 - winit::platform_impl::platform::sticky_exit_callback::haf758090da2a2168
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/mod.rs:736
  30:     0x5619a0fd7c25 - winit::platform_impl::platform::x11::EventLoop<T>::run_return::he31a7e6b243e0c91
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/x11/mod.rs:320
  31:     0x5619a0fda5e4 - winit::platform_impl::platform::x11::EventLoop<T>::run::ha99b5c0e888a01a6
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/x11/mod.rs:398
  32:     0x5619a10ad0a9 - winit::platform_impl::platform::EventLoop<T>::run::h681aa15b881edc53
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/mod.rs:652
  33:     0x5619a1095cd6 - winit::event_loop::EventLoop<T>::run::hafbf57f8f8021788
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/event_loop.rs:154
  34:     0x5619a108aa9d - iced_winit::application::run::heab70ef53b6227b8
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/iced_winit-0.3.0/src/application.rs:168
  35:     0x5619a0fafee9 - iced::application::Application::run::hd898964e8d07867b
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/iced-0.3.0/src/application.rs:219
  36:     0x5619a0fb0298 - iced::sandbox::Sandbox::run::h5f649436f0f963e5
                               at /home/bernardo/.cargo/registry/src/github.com-1ecc6299db9ec823/iced-0.3.0/src/sandbox.rs:145
  37:     0x5619a0ff0c7d - gitui_rs::main::h1f299dcd2f8ad962
                               at /home/bernardo/Projects/gitui-rs/src/main.rs:4
  38:     0x5619a1043e62 - core::ops::function::FnOnce::call_once::h566d860b3c4d46f2
                               at /home/bernardo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:227
  39:     0x5619a102331a - std::sys_common::backtrace::__rust_begin_short_backtrace::hd653d695acd7b216
                               at /home/bernardo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys_common/backtrace.rs:137
  40:     0x5619a1005d26 - std::rt::lang_start::{{closure}}::h91a78ca620366c30
                               at /home/bernardo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/rt.rs:66
  41:     0x5619a2552187 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::he179d32a5d10d957
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/core/src/ops/function.rs:259
  42:     0x5619a2552187 - std::panicking::try::do_call::hcb3d5e7be089b2b4
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:381
  43:     0x5619a2552187 - std::panicking::try::h7ac93b0cd56fb701
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panicking.rs:345
  44:     0x5619a2552187 - std::panic::catch_unwind::h7b40e396c93a4fcd
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/panic.rs:382
  45:     0x5619a2552187 - std::rt::lang_start_internal::h142b9cc66267fea1
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4/library/std/src/rt.rs:51
  46:     0x5619a1005cf7 - std::rt::lang_start::h0ccee5217edc70df
                               at /home/bernardo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/rt.rs:65
  47:     0x5619a0ff114a - main
  48:     0x7fc1d8093b25 - __libc_start_main
  49:     0x5619a0fae5be - _start
  50:                0x0 - <unknown>
Segmentation fault (core dumped)
```

Iced 0.2.0 works fine for me.

**Comment by traxys** [comment]
This happened to me too, but not in all situations. I use `i3` on Linux as my Window Manager, and whenever the window created does not take the screen for itself (meaning it is split with another window) the `tour` example crashes. Same thing happens when I try to fullscreen the application after it correctly launched

**Comment by yusdacra** [comment]
Rebased on master branch to get wgpu 0.8 and this happens again now...

<details>
<summary>Backtrace</summary>

```
[2mMay 27 14:52:43.735[0m [32m INFO[0m winit::platform_impl::platform::x11::window: Guessed window scale factor: 1.0833333333333333    
wgpu error: Validation Error

Caused by:
    In a RenderPass
      note: encoder = `iced_wgpu encoder`
    In a set_scissor_rect command
    Invalid ScissorRect parameters


thread 'main' panicked at 'Handling wgpu errors as fatal by default', /home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.8.1/src/backend/direct.rs:1955:5
stack backtrace:
   0: std::panicking::begin_panic
             at /nix/store/hg7r4ip7lv0x56rc30nafgmyjxmk2pd1-rust-default-1.52.1/lib/rustlib/src/rust/library/std/src/panicking.rs:519:12
   1: wgpu::backend::direct::default_error_handler
             at /home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.8.1/src/backend/direct.rs:1955:5
   2: core::ops::function::Fn::call
             at /nix/store/hg7r4ip7lv0x56rc30nafgmyjxmk2pd1-rust-default-1.52.1/lib/rustlib/src/rust/library/core/src/ops/function.rs:70:5
   3: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at /nix/store/hg7r4ip7lv0x56rc30nafgmyjxmk2pd1-rust-default-1.52.1/lib/rustlib/src/rust/library/alloc/src/boxed.rs:1560:9
   4: wgpu::backend::direct::ErrorSinkRaw::handle_error
             at /home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.8.1/src/backend/direct.rs:1942:9
   5: wgpu::backend::direct::Context::handle_error
             at /home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.8.1/src/backend/direct.rs:93:9
   6: <wgpu::backend::direct::Context as wgpu::Context>::command_encoder_end_render_pass
             at /home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.8.1/src/backend/direct.rs:1762:13
   7: core::ptr::drop_in_place<wgpu::RenderPass>
             at /nix/store/hg7r4ip7lv0x56rc30nafgmyjxmk2pd1-rust-default-1.52.1/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:187:1
   8: iced_wgpu::quad::Pipeline::draw
             at /home/patriot/.cargo/git/checkouts/iced-e55721123974948c/416c4f4/wgpu/src/quad.rs:271:13
   9: iced_wgpu::backend::Backend::flush
             at /home/patriot/.cargo/git/checkouts/iced-e55721123974948c/416c4f4/wgpu/src/backend.rs:116:13
  10: iced_wgpu::backend::Backend::draw
             at /home/patriot/.cargo/git/checkouts/iced-e55721123974948c/416c4f4/wgpu/src/backend.rs:82:13
  11: <iced_wgpu::window::compositor::Compositor as iced_graphics::window::compositor::Compositor>::draw
             at /home/patriot/.cargo/git/checkouts/iced-e55721123974948c/416c4f4/wgpu/src/window/compositor.rs:169:33
  12: iced_winit::application::run_instance::{{closure}}
             at /home/patriot/.cargo/git/checkouts/iced-e55721123974948c/416c4f4/winit/src/application.rs:350:45
  13: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /nix/store/hg7r4ip7lv0x56rc30nafgmyjxmk2pd1-rust-default-1.52.1/lib/rustlib/src/rust/library/core/src/future/mod.rs:80:19
  14: iced_winit::application::run::{{closure}}
             at /home/patriot/.cargo/git/checkouts/iced-e55721123974948c/416c4f4/winit/src/application.rs:193:24
  15: winit::platform_impl::platform::sticky_exit_callback
             at /home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/mod.rs:736:5
  16: winit::platform_impl::platform::x11::EventLoop<T>::run_return
             at /home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/x11/mod.rs:320:21
  17: winit::platform_impl::platform::x11::EventLoop<T>::run
             at /home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/x11/mod.rs:398:9
  18: winit::platform_impl::platform::EventLoop<T>::run
             at /home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/mod.rs:652:56
  19: winit::event_loop::EventLoop<T>::run
             at /home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/event_loop.rs:154:9
  20: iced_winit::application::run
             at /home/patriot/.cargo/git/checkouts/iced-e55721123974948c/416c4f4/winit/src/application.rs:168:5
  21: iced::application::Application::run
             at /home/patriot/.cargo/git/checkouts/iced-e55721123974948c/416c4f4/src/application.rs:219:16
  22: crust::main
             at ./src/main.rs:39:5
  23: core::ops::function::FnOnce::call_once
             at /nix/store/hg7r4ip7lv0x56rc30nafgmyjxmk2pd1-rust-default-1.52.1/lib/rustlib/src/rust/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

```
</details>

**Comment by hecrj** [comment]
Fixed by #818.
